File: src/Purchasable.java
public class Purchasable
{
    private int stock;
    private int barcode;
    private double price;
    private String name;

    public Purchasable(int stock, int barcode, double price, String name)
    {
        this.name = name;
        this.barcode = barcode;
        this.price = price;
        this.stock = stock;
    }

    public double getPrice()
    {
        return price;
    }

    public int getBarcode()
    {
        return  barcode;
    }

    public String getName()
    {
        return name;
    }

    public int getStock()
    {
        return stock;
    }
}

File: src/com/html/Js/Purchasable.js
class Purchasable
{
    stock;
    barcode;
    price;
    name;

    /**
     * A constructor that takes four parameters and intatiates the private
     * variables with those values.
     */
    Purchasable(stock, barcode, price, name)
    {
        this.name = name;
        this.barcode = barcode;
        this.price = price;
        this.stock = stock;
    }

    getPrice()
    {
        return price;
    }

    getBarcode()
    {
        return  barcode;
    }

    getName()
    {
        return name;
    }

    getStock()
    {
        return stock;
    }

    setStock(num)
    {
        this.stock = num;
    }
}